ambry-store/src/test/java/com/github/ambry/store/BlobStoreCompactorTest.java

```diff
@@ -2837,14 +2876,71 @@ private long reduceValidDataSizeInLogSegments(List<LogSegmentName> logSegmentsTo
   }
 
   /**
-   * Compacts the {@code segmentsUnderCompaction} and verifies sanity of store and data. Also verifies that no change
-   * occurred if no change was expected and vice versa. Ensures that reloading the log does not have effects on the
-   * integrity of the store and data.
+   * Compacts the {@code segmentsUnderCompaction} and verifies sanity of store and data.
    * @param segmentsUnderCompaction the names of the log segments under compaction.
    * @param deleteReferenceTimeMs the reference time in ms to use to decide whether deletes are valid.
    * @param changeExpected {@code true} if compaction will cause a change in size of data. {@code false} otherwise.
    * @throws Exception
    */
+  private void compactAndVerifyForContainerDeletion(List<LogSegmentName> segmentsUnderCompaction, long deleteReferenceTimeMs,
+      boolean changeExpected) throws Exception {
+    long logSegmentSizeSumBeforeCompaction = getSumOfLogSegmentEndOffsets();
+    long logSegmentCountBeforeCompaction = state.index.getLogSegmentCount();
+    long indexSegmentCountBeforeCompaction = state.index.getIndexSegments().size();
+
+    CompactionPolicy compactionPolicy = new HybridCompactionPolicy(config, time);
+    CompactionDetails details = new CompactionDetails(deleteReferenceTimeMs, segmentsUnderCompaction, null);
+
+    try {
+      compactor.compact(details, bundleReadBuffer);
+    } finally {
+      if (changeExpected) {
+        compactor.close(0);
+      }
+    }
+
+    if (compactor.closeLastLogSegmentIfQualified()) {
+      //refresh journal.
+      LogSegmentName activeSegmentName = state.index.log.getLastSegment().getName();
+      state.index.journal.cleanUpJournal(activeSegmentName);
+    }
+    compactor.closeLastLogSegmentIfQualified();
+    if (!changeExpected) {
+      assertEquals("Journal size should be cleaned up after last log segment closed",0, state.index.journal.getCurrentNumberOfEntries());
+    }
+
+    long logSegmentSizeAfterCompaction = getSumOfLogSegmentEndOffsets();
+    long logSegmentCountAfterCompaction = state.index.getLogSegmentCount();
```

Comment:
Check logSegmentCountAfterCompaction and indexSegmentCountAfterCompaction?

Reply:
Added the check for logSegmentCountAfterCompaction, and I removed indexSegmentCountAfterCompaction check since there's no need to check it.